README.md

```diff
@@ -1,3 +1,283 @@
 # LavaQueue
 
-yes, no docs lol.
\ No newline at end of file
+> [!IMPORTANT]
+> This plugin *requires* Lavalink `v4.0.5` or greater
+
+A simple queue plugin for [Lavalink](https://github.com/lavalink-devs/Lavalink) with a REST API.
+
+## Summary
+
+* [Installation](#installation)
+* [API](#api)
+* [Get Queue](#get-queue)
+* [Update Queue](#update-queue)
+* [Next Queue Track](#next-queue-track)
+* [Previous Queue Track](#previous-queue-track)
+* [Add Queue Track](#add-queue-tracks)
+* [Update Queue Track](#update-queue-tracks)
+* [Get Queue Track](#get-queue-track)
+* [Add Queue Track](#add-queue-track)
+* [Delete Queue Track(s)](#delete-queue-track(s))
+* [Move Queue Track](#move-queue-track)
+* [Get Queue History](#get-queue-history)
+* [Get Queue History Track](#get-queue-history-track)
+
+## Installation
+
+To install this plugin either download the latest release and place it into your `plugins` folder or add the following into your `application.yml`
+
+Replace x.y.z with the latest version number or short commit hash.
+
+```yaml
+lavalink:
+  plugins:
+    - dependency: "com.github.topi314.lavaqueue:lavaqueue-plugin:x.y.z"
+      snapshot: false # set to true if you want to use snapshot builds (see below)
+```
+
+Snapshot builds are available at https://maven.lavalink.dev/#/snapshots with the short commit hash as the version.
+
+---
+
+### Queue Types
+
+| Type            | Description                                        |
+|-----------------|----------------------------------------------------|
+| `normal`        | Tracks will be played in the order they are added. |
+| `repeat_track`  | A singular track will be repeatedly played.        |
+| `repeat_queue`  | The queue will repeat once it has ended.           |
+
+---
+
+### Common Types
+
+| Type                                                                          | Description                                               |
+|-------------------------------------------------------------------------------|-----------------------------------------------------------|
+| [track](https://lavalink.dev/api/rest.html#track)                             | A track object returned in API responses.                 |
+| [update_player_track](https://lavalink.dev/api/rest.html#update-player-track) | An update player track that can be sent in API requests.  |
+
+---
+
+### Queue Object
+
+| Field  | Type   | Description                             |
+|--------|--------|-----------------------------------------|
+| type   | string | the type of queue.                      |
+| tracks | array  | An array of track objects in the queue. |
```

Comment:
here you should link to the lavalink docs track object https://lavalink.dev/api/rest.html#track
type should be `array of [Track](https://lavalink.dev/api/rest.html#track) objects`